lib/Parser/rterrors.h

```diff
@@ -366,6 +366,7 @@ RT_ERROR_MSG(JSERR_NoAccessors, 5673, "Invalid property descriptor: accessors no
 RT_ERROR_MSG(JSERR_RegExpInvalidEscape, 5674, "", "Invalid regular expression: invalid escape in unicode pattern", kjstSyntaxError, 0)
 RT_ERROR_MSG(JSERR_RegExpTooManyCapturingGroups, 5675, "", "Regular expression cannot have more than 32,767 capturing groups", kjstRangeError, 0)
 RT_ERROR_MSG(JSERR_ProxyHandlerReturnedFalse, 5676, "Proxy %s handler returned false", "Proxy handler returned false", kjstTypeError, 0)
+RT_ERROR_MSG(JSERR_UnicodeRegExpRangeContainsCharClass, 5677, "", "Character classes not allowed in class ranges", kjstSyntaxError, 0)
```

Comment:
>Character classes [](start = 67, length = 17)

nit: mention RegExp in the message to make it clear we're talking about regexp and regexp character class ranges and not something to do with JS classes.

Also, this wording raises a question in my mind, does this also throw if you have `/[[acz]-q]/` or `/[[acz]-q]/` (embedded character class within character class).

Reply:
Never mind about embedded character classes. ECMAScript doesn't support these.
[https://www.regular-expressions.info/charclassintersect.html](https://www.regular-expressions.info/charclassintersect.html) (Java, Ruby)

---
In reply to: [201167325](https://github.com/Microsoft/ChakraCore/pull/5418#discussion_r201167325) [](ancestors = 201167325)

Reply:
Can we be double sure it's not in the spec?

Reply:
"Not supported in the other engines" is a pretty strong piece of evidence. In any case it's out of scope of this change.